### ISSUE TITLE
undefined is not an object (evaluating 'FirebaseModule.googleApiAvailability')

### ISSUE THREAD

**Issue by compojoom** [invertase/react-native-firebase#402]
Yes, I know! This has been reported like 100 times already... and I went through every reported issue.

My App is actually working, but it crashes in the simulator with ipad 2, ipad Retina, Iphone 4s, Iphone 5. The other simulators that I tried all work fine. 
I've followed the pod instructions and here is my pod file:
```pod
# Uncomment the next line to define a global platform for your project
platform :ios, '9.0'

target 'farmerjoe' do


pod 'Firebase/Core'
pod 'RNFirebase', :path => '../node_modules/react-native-firebase'

pod 'Firebase/Auth'
pod 'Firebase/Database'
pod 'Firebase/Storage'

  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga/Yoga.podspec'
  pod 'React', :path => '../node_modules/react-native', :subspecs => [
  'BatchedBridge', # Required For React Native 0.45.0+
  'Core',
  # Add any other subspecs you want to use in your project
  ]

	pod 'GoogleMaps'

   pod 'RSKImageCropper'
   pod 'QBImagePickerController'
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name == "react-native-google-maps"
      target.build_configurations.each do |config|
        config.build_settings['CLANG_ENABLE_MODULES'] = 'No'
      end
    end
  end
end

```

and I tried updating my pods just in case:
```
Analyzing dependencies
Fetching podspec for `RNFirebase` from `../node_modules/react-native-firebase`
Fetching podspec for `React` from `../node_modules/react-native`
Fetching podspec for `Yoga` from `../node_modules/react-native/ReactCommon/yoga/Yoga.podspec`
Downloading dependencies
Using Firebase (4.1.1)
Using FirebaseAnalytics (4.0.3)
Using FirebaseAuth (4.1.1)
Using FirebaseCore (4.0.6)
Using FirebaseDatabase (4.0.2)
Using FirebaseInstanceID (2.0.2)
Using FirebaseStorage (2.0.1)
Using GTMSessionFetcher (1.1.11)
Installing GoogleMaps 2.4.0 (was 2.1.1)
Using GoogleToolboxForMac (2.1.1)
Using QBImagePickerController (3.4.0)
Using RNFirebase (2.1.4)
Using RSKImageCropper (1.6.3)
Using React (0.47.2)
Using Yoga (0.47.2.React)
Using leveldb-library (1.18.3)
Using nanopb (0.3.8)
Generating Pods project
Integrating client project
Sending stats
Pod installation complete! There are 11 dependencies from the Podfile and 17 total pods installed.
```

in all cases it seems to crash with Ios 9.3, but at the same time - testing with 5s and 6 with ios9.3 works just fine...
Any ideas what I might have screwed up?

1. Application Target Platform: 
ios

2. Development Operating System: 
macOS Sierra

3. Build Tools:
XCode 

4. React Native version: 
0.47.2

5. RNFirebase Version: 
2.1.4

6. Firebase Module: 
database, auth, storage

**Comment by chrisbianca** [comment]
@compojoom Sorry for the delay in response on this one.  Are there any native errors in the logs at all?  I'm wondering whether it's a 32 bit issue - I believe all these devices you mention are 32 rather than 64 bit devices.

**Comment by compojoom** [comment]
Hey @chrisbianca 
Is there a way for me to find out if the simulator is 32bit or 64bit?

I had a look at the debug window and the only things I see are:
```
2017-09-24 15:34:27.953 [error][tid:com.facebook.react.JavaScript] undefined is not an object (evaluating 'FirebaseModule.googleApiAvailability')
2017-09-24 15:34:27.955 [fatal][tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: undefined is not an object (evaluating 'FirebaseModule.googleApiAvailability')
2017-09-24 15:34:27.964 [error][tid:com.facebook.react.JavaScript] Module AppRegistry is not a registered callable module (calling runApplication)
2017-09-24 15:34:27.978 [fatal][tid:com.facebook.react.ExceptionsManagerQueue] Unhandled JS Exception: Module AppRegistry is not a registered callable module (calling runApplication)
```

I tested the app on a 4s device and it works just fine. So, maybe it is a simulator issue, but I don't have any of the other devices to test...

**Comment by lel1234** [comment]
Had this issue on Android, using a real device.

I just run the command `react-native run-android`, it downloaded a bunch of dependencies and it ended up with an error, something like adb bridging failed, I ran it again, it gave me another error, and so on after a couple runs the build worked and the error disappeared.

TL;DR: Running `react-native run-android` 5 or 6 times fixed this error for me on Android.

**Comment by chrisbianca** [comment]
@compojoom As part of 3.0 we've removed the need to use pods for React and RNFirebase.  I'm going to close this, but please re-open if there are still issues with the new installation steps.
